File: apps/plant_shop_go/internal/models/order.go
package models

// ==============================================================================
// Importations
// ==============================================================================

import (
	"time"

	"gorm.io/gorm"
)

// ==============================================================================
// Types
// ==============================================================================

// Order represente une commande client.
type Order struct {
	ID         uint           `gorm:"primaryKey" json:"id"`
	CreatedAt  time.Time      `json:"createdAt"`
	UpdatedAt  time.Time      `json:"updatedAt"`
	DeletedAt  gorm.DeletedAt `gorm:"index" json:"-"`
	UserID     uint           `json:"userId"`
	TotalPrice float64        `json:"totalPrice"`
	Status     string         `json:"status"`
	Items      []OrderItem    `json:"orderItems,omitempty"`
}

File: apps/plant_shop_c-sharp/plant_shop_asp_dapper/Models/Order.cs
namespace plant_shop_asp_dapper.Models
{
    public class Order
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public decimal TotalPrice { get; set; }
        public string Status { get; set; } = "pending";
        public DateTime CreatedAt { get; set; }

        // Propriétés de navigation (remplies manuellement)
        public List<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}
